src/components/overview/sub-Components/imageCarousel/Zoomed Carousel/ThumbnailCircles.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import './zoomCarouselStyling.css';
+
+const ThumbnailCircles = function(props) {
+  //divide into rows to make easier to manipulate
+  let thumbnailRows = [];
+  for (
+    let index = 0;
+    index < props.state.otherImagesInStyle.length;
+    index += 4
+  ) {
+    let row = props.state.otherImagesInStyle.slice(index, index + 4);
+    thumbnailRows.push(row);
+  }
+
+  if (props.state.currentThumbnailRow !== undefined) {
+    let indexImage = props.state.currentImage;
+    let imageURL = props.state.otherImagesInStyle[indexImage].url;
+    return (
+      <div className="zoomNavigationBucket">
+        {thumbnailRows[props.state.currentThumbnailRow].map(
+          (eachThumbnailImage, index) => {
+            let selectedThumbnail;
+            if (eachThumbnailImage.url === imageURL) {
```

Comment:
This is a fantastic use of conditional code. I never thought of using conditional class naming like this. Great idea!